### PR TITLE
Add coveralls support & pre-commit hooks.

### PR DESCRIPTION
**Pull request by stephen-soltesz**
This changes moves the script commands from .travis.yml into a separate
"build" script that can be run with the pre-commit hooks.

This change also adds support for "coveralls" to collectd-mlab.

### CODE REVIEW

File: build [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -x
+
+# TODO(soltesz): update sources to support auto-discovery of tests.
+
+# Run unit tests and calculate code coverage.
+rm -f .coverage &&
```
Comment (mtlynch) [inline build]: Can we add .coverage to .gitignore?

File: build [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# Check that source has correct formatting.
+yapf --diff --recursive --style google ./ --exclude=./third_party/* &&
+
+# Run static analysis for Python bugs/cruft.
+pyflakes . &&
+
+# Run the docstring lint checker.
+PYTHONPATH=$PYTHONPATH:$PWD/third_party/docstringchecker pylint \
+    --reports=n plugin/*.py export/*.py monitoring/*.py system/vsys/*.py site-packages/mlab/disco/*.py viewer/*.py
```
Comment (mtlynch) [inline build]: Can we break these up to stay < 80 characters?

https://google.github.io/styleguide/shell.xml?showone=Line_Length_and_Long_Strings#Line_Length_and_Long_Strings

File: build (hunk 1 above)
Comment (stephen-soltesz) [inline build]: No problem. Also, adding .gitignore.

File: build (hunk 2 above)
Comment (stephen-soltesz) [inline build]: No problem. Sorry for the oversight.

**Comment by fernandalavalle** [comment]
LGTM

**Pull request merged**